### ISSUE TITLE
Expand Constant to allow getting its Type

### ISSUE THREAD

**Issue by eugenesvk** [obi1kenobi/trustfall-rustdoc-adapter#279]
I'd like to get a list of all of crate's constants with their type and values, the latest changes allowed getting the values, but still not Type.

There is a workaround you suggested using Constant ID to find the required information, but it'd be nicer to be able to use the query language for that

You mentioned that this is pending Trustfall support for types

> Once Trustfall supports custom scalar types, it gives us the option of adding a "proper" type_ field to the schema that will be more ergonomic.

**Comment by obi1kenobi** [comment]
Thanks for the issue and for the PR.

> as far as I understood, this is the ultimate crate that enables that

Thanks, it certainly tries to 😄 If you don't mind sharing, I'm curious to hear more! How did you come across this crate? What's the use case you are targeting?

**Comment by eugenesvk** [comment]
**tldr**: I wanted to parse [Windows_sys ](https://docs.rs/windows-sys/latest/windows_sys/) crate for its **constants** (names/values), started with a cargo-semver-checks crate and after some digging through sparse docs and code realized that this adapter is the one providing all the schema descriptions


A bit **more context**: I wanted to have a more ergonomic way to use Windows **constants** to call Windows **APIs** in a **scripting** language:
  - `4097`  instead of some unreadable value that you have to lookup somewhere
  - `LOCALE_SENGLANGUAGE` you could just copy&paste the full name from the API MS documentation
  - `sEngLng` or even better use a more readable shorter name


Given that there are >100k constants, I couldn't load them dynamically within the language itself, so I thought I'd write a Rust library that stores all the values and returns them from a constant name. Found the Windows_sys crate that has all the contants, realized there is no reflection in Rust (so I can't just compile the crate and ask it to list all the contants), but someone at the Rust language forums suggested to look at cargo-semver-checks which parses cargo docs that contains that information.

So I've looked at the `pub_module_level_const_missing.ron` query, realized it's also missing :( the info, digged into what the semver checks is doing and ended up at this adapter crate
